src/mon/MonitorDBStore.h

```diff
@@ -507,16 +507,12 @@ class MonitorDBStore
   }
 
   int get(const string& prefix, const string& key, bufferlist& bl) {
-    set<string> k;
-    k.insert(key);
-    map<string,bufferlist> out;
-
-    db->get(prefix, k, &out);
-    if (out.empty())
-      return -ENOENT;
-    bl.append(out[key]);
-
-    return 0;
+    bufferlist outbl;
+    int r = db->get(prefix, key, &outbl);
+    if (!r) {
+      bl.append(outbl);
+    }
```

Comment:
Only if we can assert that db->get won't totally overwrite bl (i. e. clear() + append()) and/or every case of MontitorDBStore->get() doesn't care about existing contents (if any) of passed  bl. I wasn't sure about that and decided to leave this like that.